vertx-auth-oauth2/src/main/java/io/vertx/ext/auth/oauth2/DCROptions.java

```diff
@@ -0,0 +1,52 @@
+package io.vertx.ext.auth.oauth2;
+
+import io.vertx.codegen.annotations.DataObject;
+import io.vertx.codegen.json.annotations.JsonGen;
+import io.vertx.core.http.HttpClientOptions;
+import io.vertx.core.json.JsonObject;
+
+@DataObject
+@JsonGen(publicConverter = false)
+public final class DCROptions {
+
+  /**
+   * The base url of the OIDC provider like Keycloak.
+   */
+  private String site;
+
+  /**
+   * Name of the tenant if any. Keycloak call this realm.
+   */
+  private String tenant;
+
+  /**
+   * Initial access token to authenticate with the OIDC provider.
+   */
+  private String initialAccessToken;
+
+  private HttpClientOptions httpClientOptions = new HttpClientOptions();
+
+  public DCROptions(JsonObject json) {
+    DCROptionsConverter.fromJson(json, this);
+  }
+
+  public JsonObject toJson() {
+    final JsonObject json = new JsonObject();
+    DCROptionsConverter.toJson(this, json);
+    return json;
+  }
+
+
+  public HttpClientOptions getHttpClientOptions() {
+    return httpClientOptions;
+  }
+
+  public String getResourceUri() {
+    return String.format("%s/%s/%s", site, tenant, "clients-registrations/default/");
+  }
+
+  public JsonObject getInitialAccessToken() {
```

Comment:
I get the reasons, but it feels odd, to get a initial token as a json object, it would make more sense to be just a string and do the transformation to json object at the place where it is used in the fetch request